MFVNeutralino/python/EventFilter.py

```diff
@@ -4,10 +4,10 @@ def setup_event_filter(process,
                        path_name='p',
                        trigger_filter = True,
                        trigger_filter_name = 'mfvTriggerFilter',
-                       event_filter = False,
+                       event_filter = True, #PK: used to be False
                        event_filter_jes_mult = 2,
                        event_filter_name = 'mfvEventFilter',
-                       event_filter_require_vertex = True,
+                       event_filter_require_vertex = False, #PK: used to be True
                        input_is_miniaod = False,
```

Comment:
These shouldn't need to be changed, right?

Reply:
Yep, we can leave as it is. I'll remove this change.

Reply:
Only just getting back to this now that Ang's #2 is merged into the UL branch. We should understand why this error is coming up, since we do have use cases which require the vertex in the event filter... It sounds like there's some circular logic here in the order that these are being called!

Reply:
Hi @jreic, sorry for leaving this issue unresolved until now. I have changed evenfilter.py back to the original one, and run the code. It's now working without the warning above. Indeed, it only shows up whenever the shared-jet mitigation is turned on. The result of raw vertices also implies that the split-vertices have been taken cared of: 

after resolving split-vertices: https://pekotamn.web.cern.ch/pekotamn/NtupleV27m_test_merged_vertexer_study/h_output1_vertex_paird2d.png
before resolving split-vertices:
https://pekotamn.web.cern.ch/pekotamn/NtupleV27m_test_merged_vertexer_study/h_output0_vertex_paird2d.png

Let me know if the vertexer.cc code could have been improved any further.

Reply:
Thanks Peace! I changed the small whitespace differences in EventFilter.py back, just as one less thing to worry about in the diffs.

It looks like Vertexer.cc is essentially all that is left in this PR (though it has been a while since we opened it back in October, and my memory isn't _that_ great). 

It looks like there are ~700 new lines of code added, and I'm seeing a lot of whitespace differences between the new and old file. For the whitespace, I should be able to do some relatively simple vim commands to be able to compare a bit better (since Github's diff is essentially showing that ~all of the lines of the original file have been modified!). 

But for the rest: could you confirm that this is a "minimal working" version of the updated vertexer? i.e. that there is nothing left in the new version of the file that is unused? Of course there is e.g. the track refinement (which is okay to leave in, in case it needs to be studied in the future), but do you think everything else is fairly crucial for the merging + shared jet mitigation? I assume the answer is largely yes, but I wanted to confirm before diving in. Thanks!